effekt/shared/src/main/scala/effekt/RecursiveDescent.scala

```diff
@@ -1441,7 +1455,40 @@ class RecursiveDescent(positions: Positions, tokens: Seq[Token], source: Source)
 
   // Positions
 
-  inline def nonterminal[T](inline p: => T): T = {
+  // the "span" effect used to compute the positions of nodes
+  inline def span(): Span =
+    span(_start.value)
+
+  // creates a Span with a given start and the end position of the previous token
+  inline def span(start : Int ): Span =
+    val end = previous.end + 1 // since positions by lexer are inclusive, but kiama is exclusive
```

Comment:
I noticed a change in lexer, does this still hold?

https://github.com/effekt-lang/effekt/pull/957/files#diff-b9d57417cf04b760fc643822a7f0cc8fb0827916fa671790b7f35a10f5e4774aL316-R317

Reply:
Yes for the lexer we only changed the position of EOF token. Previously, the EOF token had the position `source.content.length + 1`. This led to some spans extend outside the actual file contents. Now it has position `source.content.length = current`, which seems correct to me. The line in the parser just changes from `[start, end]` intervals to `[start, end)` intervals.